modules/core/src/main/java/org/apache/synapse/registry/Registry.java

```diff
@@ -147,4 +147,12 @@
      * @param entryKey The registry entry
      */
     public Properties getResourceProperties(String entryKey);
+
+    /**
+     * Check if a resource exists
+     *
+     * @param path The path the of resource
+     * @return true if the resource exists
+     */
+    public boolean isExists(String path);
```

Comment:
Shall we rename this function more self-descriptive isResourceExists

Reply:
Done